src/visualization/types/Scatter/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {FC, useState} from 'react'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+import {useVisXDomainSettings} from 'src/visualization/utils/useVisDomainSettings'
+import {Table} from '@influxdata/giraffe'
+interface Props {
+  setDomain: (axis: string, xDomain: [number, number]) => void
+  xDomain: number[] | null
+  table: Table
+}
+const isValidXDomain = xDomain => {
+  return (
+    xDomain &&
+    xDomain.length === 2 &&
+    typeof xDomain[0] === 'number' &&
+    typeof xDomain[1] === 'number'
+  )
+}
+export const TimeDomainAutoToggle: FC<Props> = ({
+  setDomain,
+  xDomain,
+  table,
+}) => {
+  const [isActive, setIsActive] = useState<boolean>(false)
+  const [xDom] = useVisXDomainSettings(
```

Comment:
think about what the value this variable contains represents within our system.

`xDom` doesn't mean anything. you can do better than `xDom`.